.github/contributing.md

```diff
@@ -0,0 +1,245 @@
+
+<br>
+
+## Pre-committing to Github
+
+Before committing our code to Github, there are a few considerations we need to take into account
+
+- Final code optimization and examination
+- Linting and formatting
+- Testing
+- Documentation
+
+To enforce a uniform standard of code across our codebase, we have implemented CI and pre-commit hooks to perform these operations on our code according to predefined rules and commonly utilised ruleset settings.
+
+CI Github workflows and use of pre-commit hooks by `Husky` make this possible.
+
+- A final review of your code should be performed to reduce code review overhead by other contributors
+
+  Aspects of code you should check include:
+
+    - Newlines on last line of all files
+    - Trailing spaces and log statements are removed
+    - Necessary types and documentation are present
+    - Globalization of all textual content
+    - Explicit conditional checks are utilised
+    - All edge cases along with happy paths are considered
+
+<br>
+
+- Linting and formatting is language dependent.
+    <br>
+
+    __PHP__
+
+      Formatter: PHP-CS-Fixer
+      Linter: Psalm
+
+    __JS__
+
+      Formatter: Prettier
+      Linter: ESLint
+
+
+<br>
+
+
+- Testing is the cornerstone of producing operative code.
+
+  Every commit to the `master` branch must be tested before merge.
+
+  Include the appropriate test cases for your implemented code.
+
+  This might include:
+
+  - Manual flow tests
+  - Unit tests
+  - MLT(integration) tests
+  - E2E tests
+
+  __Manual Flow Tests__: These are tests performed by contributor by physically walking through the manual process you are working upon to ensure no edge cases and breaking points are present.
+
+  __Unit Tests__: These are tests that focus on individual components or functions in isolation. They are typically low-level tests.
+
+  __Mid-Level Tests (Integration Tests)__: These tests focus on the integration between different components or modules. They ensure that these components work together as expected. These could be considered mid-level tests.
+
+  __End-to-End Tests__: These tests simulate real-world scenarios and interactions, typically spanning multiple components or even the entire application. They can be considered higher-level tests.
+
+
+<br>
+
+- Documentation is the last consideration that should be taken very seriously before committing code changes to the codebase.
+
+  - All types and fns where necessary should be properly documented.
+  Inherently, these comments need to follow `Docblock` or `Psalm` comment standards.
+
+  - No unnecessary comments or `@` tags  should be visible
+
+  - Only avail of docblock syntax to further document types when language doesn't allow concise type declarations.
+
+
+<br>
+
+
+## PR creation and management
+
+To add code to our codebase after following all above steps, you *must* create a new PR for your
+changes.
+
+The destination branch normally is `dev`, ensure this is the case.
```

Comment:
> The destination branch *must* be `dev`, ensure this is the case.

Reply:
it could be some other branch, in specific cases

Reply:
Exactly, that's what I was implying.
For example, in my reviews PR's there destination branch is `PROJ/#5/reviews-integration` as implementing large feature before merge to `dev`.

Reply:
Yes, I know, but I think "must" is a better word and we can add a note on exceptions afterwards. "normally" becomes "do whatever you want" :smile:

Reply:
I think we can resolve this by saying something like `integrating new code into the codebase could be done only by a PR toward the dev branch`.

So you can open PRs to other branches, if you need to for some reason, but in the end you need to pass through a PR towards `dev`